src/app/dataset-block/metadata-block/components/event-details/components/common/base-property/base-property.component.ts

```diff
@@ -0,0 +1,9 @@
+import { Component, Input } from "@angular/core";
+
+@Component({
+    template: "",
+})
+export abstract class BasePropertyComponent {
+    @Input() public data: unknown;
+    @Input() public class?: string;
```

Comment:
Logically, `class` property rather belongs to `SimplePropertyComponent` than here.